src/basic/recurse-dir.c

```diff
@@ -132,6 +133,18 @@ int readdir_all(int dir_fd,
         return 0;
 }
 
+int readdir_all_by_path(int fd, const char *path, RecurseDirFlags flags, DirectoryEntries **ret) {
```

Comment:
Can we name this `readdir_all_at()`? I very much prefer that we use the same naming scheme for all these functions that work in an openat() style.